eslint.config.mjs

```diff
@@ -0,0 +1,188 @@
+import js from '@eslint/js'
+import stylisticJs from '@stylistic/eslint-plugin-js'
+import stylisticTs from '@stylistic/eslint-plugin-ts'
+import globals from 'globals'
+import tseslint from 'typescript-eslint'
+import header from 'eslint-plugin-header'
+import eslintPluginPrettierRecommended from 'eslint-plugin-prettier/recommended'
+
+export default [
+	js.configs.recommended,
+	eslintPluginPrettierRecommended,
+	...tseslint.configs.recommendedTypeChecked,
+	{
+		files: ['**/*.{ts,tsx,mts,cts,js,jsx,mjs,cjs}'],
+		languageOptions: {
+			globals: {
+				...globals.browser,
+			},
+			parser: tseslint.parser,
+			parserOptions: {
+				ecmaVersion: 2020,
+				project: true,
+			},
+		},
+		plugins: {
+			'@stylistic/js': stylisticJs,
+			'@stylistic/ts': stylisticTs,
+			'@typescript-eslint': tseslint.plugin,
+			'header': header,
+		},
+		rules: {
+			'prettier/prettier': 'error',
+			'@stylistic/js/lines-between-class-members': ['error', 'always'],
+			'@stylistic/js/space-before-blocks': 'error',
+
+			'@stylistic/ts/member-delimiter-style': [
+				'error',
+				{
+					multiline: { delimiter: 'none', requireLast: true },
+					singleline: { delimiter: 'semi', requireLast: false },
+				},
+			],
+			'@typescript-eslint/explicit-function-return-type': 'off',
+			'@typescript-eslint/interface-name-prefix': 'off',
+			// '@typescript-eslint/member-ordering': [
```

Comment:
Member ordering is useful for minimizing diff size in PRs. I will enable it in the upcoming PR.